platform/xbox/crt0.c

```diff
@@ -46,8 +47,11 @@ static int main_wrapper ()
     return retval;
 }
 
-void WinMainCRTStartup ()
+void __attribute__((no_stack_protector)) WinMainCRTStartup ()
 {
+    // The security cookie needs to be initialized as early as possible, and from a non-protected function
+    __security_init_cookie();
```

Comment:
This feels like something that must be done in each new thread, so I'm surprised to see it here.

Reply:
The stack cookie is global and can only be initialized once. If you initialized it in each thread (and `__security_init_cookie()` didn't ignore multiple initialization attempts), it would screw up the stack smashing checks in all other threads.